File: hjxarchive/프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    sets = {}
    for cth in clothes:
        name = cth[0]
        type = cth[1]
        if type in sets.keys():
            sets[type].append(name)
        else:
            sets[type] = [name]
    for type in sets.keys():
        answer = answer*(len(sets[type]) + 1)
    answer -= 1
    return answer

File: tomchaccom/프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> mp;
    
    for(auto &p : clothes){
        mp[p[1]]++;
    }
    
    for(auto x : mp){
        answer*= (x.second + 1);
        
    }
    
    
    return answer -1 ;
}